packages/zoltan2/src/algorithms/connectivity/Zoltan2_VtxLabel.hpp

```diff
@@ -0,0 +1,378 @@
+#include "Tpetra_Core.hpp"
+#include "Teuchos_RCP.hpp"
+#include "Tpetra_Import.hpp"
+#include "Tpetra_FEMultiVector.hpp"
+#include "Tpetra_MultiVector_decl.hpp"
+#include "Tpetra_MultiVector_def.hpp"
+
+#include <string>
+#include <sstream>
+#include <iostream>
+#include <queue>
+#include <vector>
+
+#ifndef ZOLTAN2_VTXLABEL_
+#define ZOLTAN2_VTXLABEL_
+
+namespace Zoltan2{
+        
+        template<typename MAP>
+        class iceSheetPropagation;
+	//Graph datastructure that represents the local
+	//graph in ice sheet propagation problems
+        template<typename lno_t>
+	class icePropGraph {
+          public:
+            //n represents the number of vertices in the local graph
+            size_t n;
+            //m represents the number of edges in the local graph
+            size_t m;
+            //out_array is the adjacency array for each vertex
+            lno_t* out_array;
+            //out_degree_list is the offset array for indexing the out_array
+            lno_t* out_degree_list;
+
+            //member function to return the degree of a given vertex.
+            lno_t out_degree(lno_t vert){
+              return (out_degree_list[vert+1] - out_degree_list[vert]);
+            }
+            //member function to return the neighbors for a given vertex
+            lno_t* out_vertices(lno_t vert){
+              return (&out_array[out_degree_list[vert]]);
+            }
+        }; 
+        
+
+        //Enum that denotes
+	enum IcePropGrounding_Status {ICEPROPGS_FULL=2, ICEPROPGS_HALF=1, ICEPROPGS_NONE = 0};
+	// Struct representing a vertex label.
+	// We define our own "addition" for these labels.
+	// Later, we'll create a Tpetra::FEMultiVector of these labels.
+	template<typename lno_t, typename gno_t>
+	class IcePropVtxLabel {
+	public: 
+          //The local ID for the vertex represented by this label
+          lno_t id;
+          
+          //the global ID of a grounded vertex
+	  gno_t first_label;
+          
+          //the global ID of the vertex that sent the label
+	  gno_t first_sender;
+          
+          //this field indicates whether or not the first label is currently set
+          //(this is necessary for unsigned global types)
+          bool  first_used;
+          
+          //the global ID of a second grounded vertex
+	  gno_t second_label;
+
+          //the global ID of the vertex that sent the second identifier
+	  gno_t second_sender;
+
+          //this field indicates whether of not the second label is currently set
+          //(this is necessary for unsigned global types)
+          bool  second_used;
+
+          //this field may potentially be used to find biconnected components
```

Comment:
Are we not sure whether bcc_name will be used to find BCCs?
Or are you saying that this field is needed to find BCCs?

Reply:
This field is needed to find BCCs. I think this is an older comment, I can update it to be more accurate.